skypy/galaxy/redshift.py

```diff
@@ -5,243 +5,216 @@
-        than high. The default value is 2.0
-    size : int or tuple of ints, optional
-        The number of redshifts to sample. If None one value is sampled.
-    absolute_magnitude_max : float or scalar, optional
-        Upper limit of the considered absolute magnitudes of the galaxies to
-        wanna sample.
-    resolution : int, optional
-        Characterises the resolution of the sampling. Default is 100.
+    redshift : array_like
+        Input redshift range at which the Schechter function parameters are
+        evaluated. Galaxies are sampled over this redshift range.
+    M_star : array_like or function
+        Characteristic absolute magnitude of the Schechter function. Can be a
+        single value, an array of values for each `redshift`, or a function of
+        redshift.
+    phi_star : array_like or function
+        Normalisation of the Schechter function. Can be a single value, an
+        array of values for each `redshift`, or a function of redshift.
+    alpha : array_like or function
+        Schechter function power law index. Can be a single value, an array of
+        values for each `redshift`, or a function of redshift.
+    m_lim : float
+        Limiting apparent magnitude.
+    fsky : array_like
+        Sky fraction over which galaxies are sampled.
+    cosmology : Cosmology, optional
+        Cosmology object to convert apparent to absolute magnitudes. If not
+        given, the default cosmology is used.
+    noise : bool, optional
+        Poisson-sample the number of galaxies. Default is `True`.
 
     Returns
     -------
-    redshift_sample : ndarray or float
-        Drawn redshifts from the marginalised Schechter luminosity function.
-
-    Notes
-    -----
-    The Schechter luminosity function is given as
+    redshifts : array_like
+        Redshifts of the galaxy sample described by the Schechter luminosity
+        function.
 
-    .. math::
+    Examples
+    --------
+    Compute the number density of galaxies with redshifts between 0 and 5
+    for typical values of the "blue" galaxy luminosity function above an
+    apparent magnitude cut of 22 for a survey of 1 square degree = 1/41253 of
+    the sky.
+
+    >>> from skypy.galaxy.redshift import schechter_lf_redshift
+    >>> z = [0., 5.]
+    >>> M_star = -20.5
+    >>> phi_star = 3.5e-3
+    >>> alpha = -1.3
+    >>> z_gal = schechter_lf_redshift(z, M_star, phi_star, alpha, 22, 1/41253)
 
-        \Phi(L, z) = \frac{\Phi_\star(z)}{L_\star(z)}
-            \left(\frac{L}{L_\star(z)}\right)^\alpha
-            \exp\left(-\frac{L}{L_\star(z)}\right) \;.
+    '''
 
-    Here the luminosity is defined as
+    # compute lower truncation of scaled Schechter random variable
+    lnxmin = m_lim - cosmology.distmod(np.clip(redshift, 1e-10, None)).value
+    lnxmin -= M_star
+    lnxmin *= -0.92103403719761827361
 
-    .. math::
+    # gamma function integrand
+    def f(lnx, a):
```

Comment:
The lower limit is given by an apparent magnitude cut now, the numerical integration is only because there is no good general way to compute the upper incomplete gamma function otherwise (that I know of).

Reply:
Yes. But before we calculated it for only one limit and got the redshift dependence by multiplying with `L_*` and now we also get z dependent limits, right?